src/Foundatio.Repositories.Elasticsearch/Configuration/ElasticConfiguration.cs

```diff
@@ -32,18 +32,19 @@ public class ElasticConfiguration : IElasticConfiguration
     private readonly Lazy<ICustomFieldDefinitionRepository> _customFieldDefinitionRepository;
     protected readonly bool _shouldDisposeCache;
 
-    public ElasticConfiguration(IQueue<WorkItemData> workItemQueue = null, ICacheClient cacheClient = null, IMessageBus messageBus = null, ILoggerFactory loggerFactory = null)
+    public ElasticConfiguration(IQueue<WorkItemData> workItemQueue = null, ICacheClient cacheClient = null, IMessageBus messageBus = null, TimeProvider timeProvider = null, ILoggerFactory loggerFactory = null)
     {
         _workItemQueue = workItemQueue;
+        TimeProvider = timeProvider ?? TimeProvider.System;
         LoggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
         _logger = LoggerFactory.CreateLogger(GetType());
         Cache = cacheClient ?? new InMemoryCacheClient(new InMemoryCacheClientOptions { LoggerFactory = loggerFactory, CloneValues = true });
-        _lockProvider = new CacheLockProvider(Cache, messageBus, loggerFactory);
+        _lockProvider = new CacheLockProvider(Cache, messageBus, null, loggerFactory);
```

Comment:
Still think this can be pulled off the cache client.

Reply:
It is. Look in CacheLockProvider.

Reply:
Updated